crates/cast/src/starknet_commands/declare.rs

```diff
@@ -130,29 +130,28 @@ pub async fn declare(
         .iter()
         .find_map(|contract| {
             if contract.contract_name == contract_name {
-                return Some(contract.artifacts.sierra.clone());
+                return contract.artifacts.sierra.clone();
             }
             None
         })
-        .unwrap_or_else(|| {
-            panic!("Failed to find contract {contract_name} in starknet_artifacts.json")
+        .ok_or_else(|| -> anyhow::Result<Utf8PathBuf> {
+            anyhow::bail!("Failed to find contract {contract_name} in starknet_artifacts.json")
         });
-    let sierra_contract_path = &compiled_directory.join(sierra_path);
+    let sierra_contract_path = &compiled_directory.join(sierra_path.expect("Cannot find sierra artifact file - please make sure sierra is set to 'true' under your [[target.starknet-contract]] field in Scarb.toml"));
```

Comment:
I think we should not `expect` here, because it will panic on error. It would be nicer to just print
```
command: declare
error: Cannot find sierra
```

Reply:
We can achieve this simply by mapping en error and returning `bail!(...`

Reply:
as discussed, will leave as is for now (to be improved in https://github.com/foundry-rs/starknet-foundry/issues/352)